docs/contracts/errors.rst

```diff
@@ -0,0 +1,72 @@
+.. index:: ! error, revert
+
+.. _errors:
+
+*******************************
+Errors and the Revert Statement
+*******************************
+
+Errors in Solidity provide a convenient and gas-efficient way to explain to the
+user why an operation failed. They can be defined inside and outside of contracts (including interfaces and libraries).
+
+They have to be used together with the :ref:`revert statement <revert-statement>`
+which causes
+all changes in the current call to be reverted and passes the error data back to the
+caller.
+
+::
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity ^0.8.3;
+
+    /// Insufficient balance for transfer. Needed `required` but only
+    /// `available` available.
+    /// @param available balance available.
+    /// @param required requested amount to transfer.
+    error InsufficientBalance(uint256 available, uint256 required);
+
+    contract TestToken {
+        mapping(address => uint) balance;
+        function transfer(address to, uint256 amount) public {
+            if (amount > balance[msg.sender])
+                revert InsufficientBalance({
+                    available: balance[msg.sender],
+                    required: amount
+                });
+            balance[msg.sender] -= amount;
+            balance[to] += amount;
+        }
+        // ...
+    }
+
+Errors cannot be overloaded or overridden but are inherited.
+The same error can be defined in multiple places as long as the scopes are distinct.
+Instances of errors can only be created in ``revert`` statements.
```

Comment:
Maybe `...only be created using revert statements.`  sound better

Reply:
Yeah, it would not hurt to state more explicitly that the name of the contract containing the error is not part of its signature. It's the same for events but still, I think it's natural to expect some namespacing here so if there's none we should underscore that fact.